bears/typescript/TSLintBear.py

```diff
@@ -26,16 +28,171 @@ class TSLintBear:
+        if not tslint_config:
+            _tslint_directive_selector_naming_convention = (
+                naming_convention_map.get(
+                   directive_selectors_naming_convention, ''))
+            _tslint_component_selector_naming_convention = (
+                naming_convention_map.get(
+                   component_selectors_naming_convention, ''))
+            rules = {
+                     "directive-selector-name": [
+                        _tslint_directive_selector_naming_convention is not '',
+                        _tslint_directive_selector_naming_convention],
+                     "component-selector-name": [
+                        _tslint_component_selector_naming_convention is not '',
+                        _tslint_component_selector_naming_convention],
+                     "directive-selector-type":
+                        [check_directive_selectors_type,
+                         directive_selectors_type],
+                     "component-selector-type":
+                        [check_component_selectors_type,
+                         component_selectors_type],
+                     "directive-selector-prefix":
+                        [force_directive_selectors_prefix,
+                         directive_selectors_prefix],
+                     "component-selector-prefix":
+                        [force_component_selectors_prefix,
+                         component_selectors_prefix],
+                     "use-input-property-decorator":
+                         use_input_property_decorators,
+                     "use-output-property-decorator":
+                         use_output_property_decorators,
+                     "use-host-property-decorator":
+                         use_host_property_decorators,
+                     "no-attribute-parameter-decorator":
+                         allow_attribute_parameter_decorators,
+                     "no-input-rename": allow_input_rename,
+                     "no-output-rename": allow_output_rename,
+                     "no-forward-ref": allow_forward_references,
+                     "use-life-cycle-interface": use_life_cycle_interface,
+                     "use-pipe-transform-interface":
+                         use_pipe_transform_interface,
+                     "pipe-naming":
+                        [force_pipe_prefix, pipe_naming_convention,
+                         pipe_prefix],
+                     "component-class-suffix": component_class_suffix,
+                     "directive-class-suffix": directive_class_suffix,
+                     "import-destructuring-spacing":
+                         check_import_destructuring_spacing
+                    }
+            configs = {"rulesDirectory": "node_modules/codelyzer",
+                       "rules": rules}
+            with prepare_file('', 'lint.json') as (file_, fname):
+                json.dump(configs, open(fname, 'w'))
+                args += ('--config', fname)
```

Comment:
uhm isn't the generate_config function made for that it already does this file writing for you? Also I think the file gets destroyed when exiting the `with`